opal/mca/btl/tcp/btl_tcp_endpoint.c

```diff
@@ -406,12 +397,30 @@ mca_btl_tcp_endpoint_send_blocking(mca_btl_base_endpoint_t* btl_endpoint,
 
 static int mca_btl_tcp_endpoint_send_connect_ack(mca_btl_base_endpoint_t* btl_endpoint)
 {
-    /* send process identifier to remote endpoint */
+
     opal_process_name_t guid = opal_proc_local_get()->proc_name;
 
```

Comment:
I don't see the benefit of having a unique magic_id_string, it can be easily generated. The original idea was to exchange a unique key only known by the processes in the same jobid (or by those where the modex info has been delivered). I don't recall it's name, but @rhc54 can certainly help here.

By doing this with a constant string in 2 steps the overhead of the connection establishment increases unnecessary.